lib/guardian.rb

```diff
@@ -232,7 +232,7 @@ def can_grant_badges?(_user)
   end
 
   def can_delete_reviewable_queued_post?(reviewable)
-    reviewable.present? && authenticated? && reviewable.created_by_id == @user.id
+    reviewable.present? && authenticated? && (reviewable.created_by_id == @user.id || @user.admin?)
```

Comment:
Totally stylistic but might be more readable with guards

```
def can_delete_reviewable_queued_post?(reviewable)
    return false if reviewable.blank?
    return false if !authenticated?
    return true if @user.admin?
    
    reviewable.created_by_id == @user.id
  end
```

Reply:
https://github.com/discourse/discourse/pull/21188